proxy/ProxyClientSession.cc

```diff
@@ -182,29 +167,26 @@ void
 ProxyClientSession::do_api_callout(TSHttpHookID id)
 {
   ink_assert(id == TS_HTTP_SSN_START_HOOK || id == TS_HTTP_SSN_CLOSE_HOOK);
-
-  this->api_hookid  = id;
-  this->api_scope   = API_HOOK_SCOPE_GLOBAL;
-  this->api_current = nullptr;
-
-  if (this->hooks_on && this->has_hooks()) {
-    SET_HANDLER(&ProxyClientSession::state_api_callout);
-    this->state_api_callout(EVENT_NONE, nullptr);
-  } else {
-    this->handle_api_return(TS_EVENT_HTTP_CONTINUE);
+  if (hooks_on) {
+    hook_state.init(id, http_global_hooks, &api_hooks);
+    cur_hook = hook_state.getNext();
```

Comment:
Add a comment that `getNext` is called to verify whether there are any hooks at all to invoke.